code/controllers/subsystem/non_firing/SSlate_mapping.dm

```diff
@@ -27,4 +27,30 @@ SUBSYSTEM_DEF(late_mapping)
 		var/duration = stop_watch(watch)
 		log_startup_progress("Generated [mgcount] mazes in [duration]s")
 
+	maintenance_mice()
+
 	GLOB.spawn_pool_manager.process_pools()
+
+/**
+ * Randomly spawns mice in maintenance instead of being purely fixed spawn points
+ */
+/datum/controller/subsystem/late_mapping/proc/maintenance_mice()
+	var/watch = start_watch()
+	log_startup_progress("Populating maintenance with mice...")
+
+	// Looking up for maintenance floors specifically as possible spawn points
+	var/list/maintenance_turfs = list()
+	for(var/area/station/maintenance/A in SSmapping.existing_station_areas)
+		for(var/turf/simulated/floor/F in A)
+			if(locate(/obj/structure/window) in F)
+				continue
```

Comment:
IIRC, it will be quicker to iterate through all the tiles on the z-level instead, but you may wish to measure both performances
```suggestion
	for(var/area/station/maintenance/A in SSmapping.existing_station_areas)
	for(var/turf/simulated/floor/F in block(1, 1, level_name_to_num(MAIN_STATION), world.maxx, world.maxy))
		if(!istype(loc, /area/station/maintenance))
			continue
		if(locate(/obj/structure/window) in F)
			continue
```